src/base/hide_password.cpp

```diff
@@ -0,0 +1,861 @@
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+};
+
+#define SKIP_SPACE_CHARACTERS(p)                \
+  do {                                          \
+        while (is_space[(unsigned char) *p])     \
+	    {                                   \
+	      p++;                              \
+	    }                                   \
+   } while(0)
+
+#define MAX_PWD_LENGTH                   (0x0FFFFFFF)
+#define SET_PWD_LENGTH(s, e)             ((e) - (s))
+#define SET_PWD_ADDINFO(comma, en_pwd)      (((comma) ? (0x01 << 30) : 0) | (((en_pwd) & 0x03) << 28))
+#define SET_PWD_LENGTH_N_ADDINFO(s, e, comma, en_pwd)  (SET_PWD_ADDINFO((comma), (en_pwd)) | SET_PWD_LENGTH((s), (e)))
+#define IS_PWD_NEED_COMMA(pwd_info_ptr)    (((pwd_info_ptr)[1] >> 30) & 0x01)
+#define IS_PWD_NEED_PASSWORD(pwd_info_ptr) ((EN_ADD_PWD_STRING)(((pwd_info_ptr)[1] >> 28) & 0x03))
+#define GET_END_PWD_OFFSET(pwd_info_ptr)   ((pwd_info_ptr)[0] + ((pwd_info_ptr)[1] & MAX_PWD_LENGTH))
+
+class hide_password
+{
+  private:
+    char *skip_comment_string (char *query);
+    char *get_quot_string (char *ps);
+    char *get_token (char *&in, int &len);
+    char *get_method_passowrd_start_position (char *ps, char *method_name, int *method_password_len);
+    char *get_passowrd_pos_n_len (char *query, bool is_create, bool is_server, int *password_len,
+				  bool *is_found_pwd_keyword);
+    char *skip_one_query (char *query);
+    bool check_lead_string_in_query (char **query, char **method_name, bool *is_create, bool *is_server);
+    void fprintf_replace_newline (FILE *fp, char *query, int (*cas_fprintf) (FILE *, const char *, ...));
+    bool check_capitalized_keyword_create (char *query);
+    const char *get_password_string (char *qryptr, int *pwd_info_ptr);
+
+    bool  m_use_backslash_escape;
+
+  public:
+    hide_password ()
+    {
+      m_use_backslash_escape = ! prm_get_bool_value (PRM_ID_NO_BACKSLASH_ESCAPES);
+    };
+    ~hide_password ()
+    {
+    };
+
+    void find_password_positions (char *query, HIDE_PWD_INFO_PTR hide_pwd_ptr);
+    int snprint_password (char *msg, int size, char *query, HIDE_PWD_INFO_PTR hide_pwd_ptr);
+    void fprintf_password (FILE *fp, char *query, HIDE_PWD_INFO_PTR hide_pwd_ptr,
+			   int (*cas_fprintf) (FILE *, const char *, ...));
+};
+
+char *
+hide_password::skip_comment_string (char *query)
+{
+  query += 2;
+  if (query[-1] != '*')
+    {
+      // case) -- or //
+      while (*query)
+	{
+	  if (*query == '\n')
+	    {
+	      query++;
+	      break;
+	    }
+	  query++;
+	}
+    }
+  else
+    {
+      // case)  /*
+      while (*query)
+	{
+	  if (*query == '*' && query[1] == '/')
+	    {
+	      query += 2;
+	      break;
+	    }
+	  query++;
+	}
+    }
+  return query;
+}
+
+char *
+hide_password::get_quot_string (char *ps)
```

Comment:
'**get_quot_string**' seems to recognize the quoted string itself.
Actually, this function is understood as a function that finds the end of the quote string.
Litte bit confused, how about changing the name, for example `find_end_of_quot ()`

Reply:
What you thought is correct.
I think the function name you suggested is appropriate.